zha/application/platforms/switch.py

```diff
@@ -140,6 +141,70 @@ def handle_cluster_handler_attribute_updated(
             self.maybe_emit_state_changed_event()
 
 
+@STRICT_MATCH(cluster_handler_names=CLUSTER_HANDLER_BINARY_OUTPUT)
+class BinaryOutputSwitch(PlatformEntity, BaseSwitch):
+    """BinaryOutputCluster switch."""
+
+    def __init__(
+        self,
+        cluster_handlers: list[ClusterHandler],
+        endpoint: Endpoint,
+        device: Device,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(cluster_handlers, endpoint, device, **kwargs)
+        self._binary_output_cluster_handler: BinaryOutputClusterHandler = (
+            self.cluster_handlers[CLUSTER_HANDLER_BINARY_OUTPUT]
+        )
+
+    def _is_supported(self) -> bool:
+        if self._binary_output_cluster_handler.description is None:
+            return False
+
+        return super()._is_supported()
+
+    def recompute_capabilities(self) -> None:
+        """Recompute capabilities."""
+        super().recompute_capabilities()
+        self._attr_fallback_name = self._binary_output_cluster_handler.description
+
+    def on_add(self) -> None:
+        """Run when entity is added."""
+        super().on_add()
+        self._on_remove_callbacks.append(
+            self._binary_output_cluster_handler.on_event(
+                CLUSTER_HANDLER_ATTRIBUTE_UPDATED,
+                self.handle_cluster_handler_attribute_updated,
+            )
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the switch is on."""
+        if self._binary_output_cluster_handler.present_value is None:
+            return False
+        return bool(self._binary_output_cluster_handler.present_value)
```

Comment:
Now that we use `bool(.present_value)`, we should be able to remove the `present_value is None` check.

Reply:
True. I was keeping it around because I'm not sure if I agree with this behavior in general, it seems like `unavailable` would be a better state?

Reply:
Yeah, I also thought about that. It makes sense for sensors and so on, but I'm not sure how this behaves with switches. Can you still turn on a switch if it's unavailable?